test/player_integration.js

```diff
@@ -72,6 +72,31 @@ describe('Player', () => {
     });
   });  // describe('attach')
 
+  describe('destroy', () => {
+    afterEach(() => {
+      compiledShaka.util.Timer.onTimerTick = null;
+    });
+
+    // Regression test for:
+    // https://github.com/shaka-project/shaka-player/issues/4850
+    it('does not leave any lingering timers', async () => {
+      // Play the video for a little while.
+      await player.load('test:sintel_compiled');
+      video.play();
+      await waiter.waitUntilPlayheadReachesOrFailOnTimeout(video, 1, 10);
+
+      // Destroy the player.
+      await player.destroy();
+
+      // Wait a while to see if anything ticks.
+      compiledShaka.util.Timer.onTimerTick = () => {
+        fail('A timer has continued to tick after the destruction of the ' +
+             'player.');
+      };
+      await shaka.test.Util.delay(3);
```

Comment:
This seems brittle, and may explain why it failed in one case on GitHub.  If a surviving timer is running less often than every 3 seconds, a failure here could be non-deterministic.

Here's an alternative:

Instead of hooking each timer firing, have a static variable that tracks live instances.  When a timer starts, add it to the set.  When it stops, remove it from the set.  No delays required.

Or better yet, make it a map, with timer as key and stack trace as value.  When a timer survives the destruction of the player, you will also be able to say where it was created.  If this tracking is contingent on DEBUG being defined, there will be no penalty in production.  (See my other comment on that.)

Reply:
Done.